azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -25,39 +25,47 @@ class MQTTTransport(object):
     :type on_mqtt_message_received: Function
     """
 
-    def __init__(self, client_id, hostname, username, ca_cert=None):
+    def __init__(self, client_id, hostname, username, ca_cert=None, x509_cert=None):
         """
         Constructor to instantiate an MQTT protocol wrapper.
         :param str client_id: The id of the client connecting to the broker.
         :param str hostname: Hostname or IP address of the remote broker.
         :param str username: Username for login to the remote broker.
         :param str ca_cert: Certificate which can be used to validate a server-side TLS connection (optional).
+        :param x509_cert: Certificate which can be used to authenticate connection to a server in lieu of a password (optional).
         """
         self._client_id = client_id
         self._hostname = hostname
         self._username = username
         self._mqtt_client = None
         self._ca_cert = ca_cert
+        self._x509_cert = x509_cert
 
         self.on_mqtt_connected = None
         self.on_mqtt_disconnected = None
         self.on_mqtt_message_received = None
 
         self._op_manager = OperationManager()
 
-        self._create_mqtt_client()
+        self._mqtt_client = self._create_mqtt_client()
 
     def _create_mqtt_client(self):
         """
         Create the MQTT client object and assign all necessary event handler callbacks.
         """
         logger.info("creating mqtt client")
 
-        self._mqtt_client = mqtt.Client(
+        # Instantiate client
+        mqtt_client = mqtt.Client(
             client_id=self._client_id, clean_session=False, protocol=mqtt.MQTTv311
         )
-        self._mqtt_client.enable_logger(logging.getLogger("paho"))
+        mqtt_client.enable_logger(logging.getLogger("paho"))
 
+        # Configure TLS/SSL
+        ssl_context = self._create_ssl_context()
```

Comment:
I removed the `mqtt_client.tls_set_insecure(False)` because it already defaults to `False`, and having the call at all in the code is more dangerous than not having it (since all it would take to enable it to change `False` to `True`). Furthermore, the API itself recommends not including it in production code for this reason.

Reply:
+1

---
In reply to: [307945405](https://github.com/Azure/azure-iot-sdk-python-preview/pull/149#discussion_r307945405) [](ancestors = 307945405)